chia/util/misc.py

```diff
@@ -75,3 +78,13 @@ def prompt_yes_no(prompt: str = "(y/n) ") -> bool:
 
 def get_list_or_len(list_in: Sequence[object], length: bool) -> Union[int, Sequence[object]]:
     return len(list_in) if length else list_in
+
+
+def dataclass_to_json_dict(instance: Any) -> Dict[str, Any]:
+    return recurse_jsonify(dataclasses.asdict(instance))
+
+
+@dataclasses.dataclass
+class KeyValue:
+    key: str
+    value: Optional[str]
```

Comment:
This is too generic a name for the specific types, I think.  Where you hinted this you would use `KeyValue[str, Optional[str]]`.

```suggestion
@dataclasses.dataclass
class KeyValue(Generic[K, V]):
    key: K
    value: V
```

https://mypy-play.net/?mypy=latest&python=3.10&gist=dde9ea576b05527136c4c74b11a9cde1
```python
import dataclasses
import typing


K = typing.TypeVar("K")
V = typing.TypeVar("V")


@dataclasses.dataclass
class KeyValue(typing.Generic[K, V]):
    key: K
    value: V


kva = KeyValue(key="abc", value=123)
reveal_type(kva)

kvb = KeyValue[str, typing.Optional[int]](key="abc", value=123)
reveal_type(kvb)

kvc: KeyValue[bool, float]
kvc = KeyValue(key=True, value=4.2)
reveal_type(kvc)
```

```
main.py:16: note: Revealed type is "__main__.KeyValue[builtins.str, builtins.int]"
main.py:19: note: Revealed type is "__main__.KeyValue[builtins.str, Union[builtins.int, None]]"
main.py:23: note: Revealed type is "__main__.KeyValue[builtins.bool, builtins.float]"
Success: no issues found in 1 source file
```

Reply:
See [bc2c789](https://github.com/Chia-Network/chia-blockchain/pull/11365/commits/bc2c7895e723240fbad0d568b7c5ea771b5b1bcd), i moved it into `farmer_rpc_api.py` and named it less generic since your approach doesn't work with `dataclass_from_dict`.